backend/database/versions/9023d0bf61ab_harmonize_schema_state_prod.py

```diff
@@ -0,0 +1,86 @@
+
+# revision identifiers, used by Alembic.
+revision = "9023d0bf61ab"
+down_revision = "d9fb29897244"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_table("dataset_deployment")
+    op.alter_column("dataset", "project_id", existing_type=sa.VARCHAR(), nullable=False)
+    op.alter_column("dataset", "project_status", existing_type=sa.VARCHAR(), nullable=False)
+    op.alter_column("dataset", "revision", existing_type=sa.INTEGER(), nullable=True)
+    op.alter_column("deployment_directory", "created_at", existing_type=postgresql.TIMESTAMP(), nullable=False)
+    op.alter_column("deployment_directory", "dataset_id", existing_type=sa.VARCHAR(), nullable=False)
+    op.alter_column("deployment_directory", "updated_at", existing_type=postgresql.TIMESTAMP(), nullable=False)
+    op.create_foreign_key(None, "deployment_directory", "dataset", ["dataset_id"], ["id"])
+    op.alter_column("project", "needs_attestation", existing_type=sa.BOOLEAN(), nullable=True)
+    op.alter_column(
+        "project",
+        "processing_state",
+        existing_type=postgresql.ENUM(
+            "NA", "IN_VALIDATION", "IN_ARTIFACT_CREATION", "IN_DEPLOYMENT", name="processingstate"
+        ),
+        nullable=True,
+    )
+    op.alter_column(
+        "project",
+        "validation_state",
+        existing_type=postgresql.ENUM("NOT_VALIDATED", "VALID", "INVALID", name="validationstate"),
+        nullable=True,
+    )
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.alter_column(
+        "project",
+        "validation_state",
+        existing_type=postgresql.ENUM("NOT_VALIDATED", "VALID", "INVALID", name="validationstate"),
+        nullable=False,
+    )
+    op.alter_column(
+        "project",
+        "processing_state",
+        existing_type=postgresql.ENUM(
+            "NA", "IN_VALIDATION", "IN_ARTIFACT_CREATION", "IN_DEPLOYMENT", name="processingstate"
+        ),
+        nullable=False,
+    )
+    op.alter_column("project", "needs_attestation", existing_type=sa.BOOLEAN(), nullable=False)
+    op.drop_constraint(None, "deployment_directory", type_="foreignkey")
+    op.alter_column("deployment_directory", "updated_at", existing_type=postgresql.TIMESTAMP(), nullable=True)
+    op.alter_column("deployment_directory", "dataset_id", existing_type=sa.VARCHAR(), nullable=True)
+    op.alter_column("deployment_directory", "created_at", existing_type=postgresql.TIMESTAMP(), nullable=True)
+    op.alter_column("dataset", "revision", existing_type=sa.INTEGER(), nullable=False)
+    op.alter_column("dataset", "project_status", existing_type=sa.VARCHAR(), nullable=True)
+    op.alter_column("dataset", "project_id", existing_type=sa.VARCHAR(), nullable=True)
+    op.create_table(
+        "dataset_deployment",
+        sa.Column("id", sa.VARCHAR(), autoincrement=False, nullable=False),
```

Comment:
I think setting autoincrement isn't necessary here (and other locations throughout this PR). It's only does something for INTEGER type Columns.

Reply:
Huh yeah I don't know why that's there. I guess if it were True it might try to make it a `SERIAL`? I feel like that just wouldn't work though.